feat(curriculum): add shortest path algorithm workshop (#61147)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-shortest-path-algorithm/687774e17e0972345de2527a.md

---
id: 687774e17e0972345de2527a
title: Step 16
challengeType: 20
dashedName: step-16
---

# --description--

If a valid node was found in the pass, you need to mark it as visited so it won't be considered again in future iterations.

After the `if current == -1:` conditional, add a line that sets `visited[current]` to `True`.

# --hints--

> You should set `visited[current]` to `True`.
> 
> ```js
> ({
>   test: () => {
>     assert(runPython(`"visited[current] = True" in _code`))
>   }
> })
> ```

Reviewer: ````suggestion
You should set `visited[current]` to `True` after your last `if` statement.

```js
({
  test: () => {
    (runPython(`
    block = _Node(_code).find_function("shortest_path").find_for_loops()[0].find_bodies()[0]
    if_stmt = """if current == -1:
      break"""
    target_stmt = "visited[current] = True"
    assert block.is_ordered(if_stmt, target_stmt)
    `))
  }
})
```
````
